src/dataflow/src/source/kafka.rs

```diff
@@ -99,14 +100,22 @@ impl SourceReader for KafkaSourceReader {
             _ => unreachable!(),
         };
 
-        let KafkaSourceConnector {
-            addrs,
+        let KafkaSourceInstance {
+            connector,
             topic,
-            config_options,
             group_id_prefix,
             cluster_id,
             ..
         } = kc;
+        let (addrs, config_options) = match connector {
+            KafkaSourceConnector::Literal(KafkaSourceConnectorLiteral {
+                addrs,
+                config_options,
+            }) => (addrs, config_options),
+            _ => panic!(
+                "Invalid KafkaSourceConnector form, only Literal is allowed in Dataflow layer"
```

Comment:
Should this just be `unreachable!` because you can't trigger this code path, or is this intimating that we plan to take references to `Connector`s into literals elsewhere?

Reply:
This is indicating that the latter should have happened, so if this code path is reached this invariant has been violated.